File: development/OnBoarding/OnBoarding/OnBoarding.Web/src/app/model/associatePlan.ts
export interface IAssociatePlan {
    Id: string,
    AssociateId: string,
    AssociateCode: string,
    AssociateName: string,
    OwnerName: string,
    ModeName: string,
    Reference: string,
    KnowledgeTransferTitle: string,
    Duration: number,
    Week: number,
    Day: number
    ScheduledDate: Date,
    ProofType: number,
    Proof: any,
    CompletionDate?: Date    
}
export class AssociatePlan implements IAssociatePlan {
    constructor(
        public Id: string,
        public AssociateId: string,
        public AssociateCode: string,
        public AssociateName: string,
        public OwnerName: string,
        public ModeName: string,
        public Reference: string,
        public KnowledgeTransferTitle: string,
        public Duration: number,
        public Week: number,
        public Day: number,
        public ScheduledDate: Date,
        public ProofType: number,
        public Proof: any,
        public CompletionDate?: Date        
        ) {

    }
}

File: development/OnBoarding/OnBoarding/OnBoarding.Contract/AssociatePlan/AssociatePlan.cs
namespace OnBoarding.Contract
{
    using System;
   public class AssociatePlan
    {
        public Guid Id { get; set; }

        public Guid AssociateId { get; set; }

        public string AssociateCode { get; set; }

        public string AssociateName { get; set; }

        public string OwnerName { get; set; }

        public string ModeName { get; set; }

        public string  Reference { get; set; }
        
        public string KnowledgeTransferTitle { get; set; }

        public decimal Duration { get; set; }

        public int Week { get; set; }

        public int Day { get; set; }

        public DateTime ScheduledDate { get; set; }

        public DateTime? CompletionDate { get; set; }

        public byte[] Proof { get; set; }

        public int ProofType { get; set; }
    }
}
